client/evm_logs.go

```diff
@@ -0,0 +1,72 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/ava-labs/avalanchego/cache"
+	"github.com/ava-labs/coreth/core/types"
+	"github.com/ava-labs/coreth/ethclient"
+	"github.com/ava-labs/coreth/interfaces"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+const (
+	logCacheSize       = 100
+	transferMethodHash = "0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef"
+)
+
+// EvmLogsClient is a client for requesting evm logs
+type EvmLogsClient struct {
+	ethClient *ethclient.Client
+	cache     *cache.LRU
+}
+
+// NewEvmLogsClient returns a new EVM Logs client
+func NewEvmLogsClient(endpoint string) (*EvmLogsClient, error) {
+	endpoint = strings.TrimSuffix(endpoint, "/")
+
+	c, err := ethclient.Dial(fmt.Sprintf("%s%s", endpoint, prefixEth))
+	if err != nil {
+		return nil, err
+	}
+
+	cache := &cache.LRU{Size: logCacheSize}
```

Comment:
I'd suggest removing this cache. During normal operation, you'll never fetch the same block twice.

Reply:
This cache is used a lot.  The cleanest way to get EVM logs was to fetch them on the transaction level.  So each transaction within a block will pull the logs from the cache